src/codes/delta.rs

```diff
@@ -101,10 +101,8 @@ impl<B: GammaWrite<BE>> DeltaWrite<BE> for B {
         &mut self,
         value: u64,
     ) -> Result<()> {
-        if USE_TABLE {
-            if delta_tables::write_table_be(self, value)? {
-                return Ok(());
-            }
+        if USE_TABLE && delta_tables::write_table_be(self, value)? {
```

Comment:
I have a problem here because USE_TABLE is really conditional compilation and mixing it with runtime code is ugly. I like that there is a clear conditional box and inside there's code. I guess there's no way to convince Clippy that a nested if with a constant value is OK, in which case I'd convince Clippy (at least for this crate) that nested if's are OK.

Reply:
BTW, this is one thing that is specifically built in Java—code in an if controlled by a constant is always considered reachable, or you couldn't do conditional compilation with if's and final static booleans.